pallets/compliance-manager/src/benchmarking.rs

```diff
@@ -0,0 +1,375 @@
+    pub fn add_default_trusted_claim_issuer(self: &Self, i: u32) {
+        make_issuers::<T>(i).into_iter().for_each(|issuer| {
+            Module::<T>::add_default_trusted_claim_issuer(
+                self.owner.origin.clone().into(),
+                self.ticker.clone(),
+                issuer,
+            )
+            .expect("Default trusted claim issuer cannot be added");
+        });
+    }
+}
+
+struct ComplianceRequirementBuilder<T: Trait> {
+    info: ComplianceRequirementInfo<T>,
+    has_been_added: bool,
+}
+
+impl<T: Trait> ComplianceRequirementBuilder<T> {
+    pub fn new(
+        trusted_issuer_count: u32,
+        sender_conditions_count: u32,
+        receiver_conditions_count: u32,
+    ) -> Self {
+        // Create accounts and token.
+        let owner = User::<T>::new("OWNER", SEED);
+        let buyer = User::<T>::new("BUYER", SEED);
+        let ticker = make_token::<T>(&owner, b"1".to_vec());
+
+        // Create issuers (i) and conditions(s & r).
+        let issuers = make_issuers::<T>(trusted_issuer_count);
+        let sender_conditions = make_conditions(sender_conditions_count, &issuers);
+        let receiver_conditions = make_conditions(receiver_conditions_count, &issuers);
+
+        let info = ComplianceRequirementInfo {
+            owner,
+            buyer,
+            ticker,
+            issuers,
+            sender_conditions,
+            receiver_conditions,
+        };
+
+        Self {
+            info,
+            has_been_added: false,
+        }
+    }
+
+    /// Register the compliance requirement in the module.
+    pub fn add_compliance_requirement(mut self: Self) -> Self {
+        assert!(
+            self.has_been_added == false,
+            "Compliance has been added before"
+        );
+        Module::<T>::add_compliance_requirement(
+            self.info.owner.origin.clone().into(),
+            self.info.ticker.clone(),
+            self.info.sender_conditions.clone(),
+            self.info.receiver_conditions.clone(),
+        )
+        .expect("Compliance requirement cannot be added");
+        self.has_been_added = true;
+        self
+    }
+
+    pub fn build(self: Self) -> ComplianceRequirementInfo<T> {
+        self.info
+    }
+}
+
+benchmarks! {
+    _ {}
+
+    add_compliance_requirement {
+        // INTERNAL: This benchmark only evaluate the adding operation. Its execution should be measured in another module.
+        let s in 1..MAX_SENDER_CONDITIONS_PER_COMPLIANCE;
+        let r in 1..MAX_RECEIVER_CONDITIONS_PER_COMPLIANCE;
+
+        let d = ComplianceRequirementBuilder::<T>::new(MAX_TRUSTED_ISSUER_PER_CONDITION, s, r).build();
+
+    }: _(d.owner.origin, d.ticker, d.sender_conditions.clone(), d.receiver_conditions.clone())
+    verify {
+        let req = Module::<T>::asset_compliance(d.ticker).requirements.pop().unwrap();
+        ensure!( req.sender_conditions == d.sender_conditions, "Sender conditions not expected");
```

Comment:
We can use asserts instead of ensures since the backtrace doesn't track the ensure error messages anyway iirc.